From 831bbcc57424dc1f81a7f39f66e28d9c6bb0b6bf Mon Sep 17 00:00:00 2001
Message-Id: <831bbcc57424dc1f81a7f39f66e28d9c6bb0b6bf.1387276076.git.minovotn@redhat.com>
In-Reply-To: <e610718166120379517e80d1a7aa12d60294209b.1387276076.git.minovotn@redhat.com>
References: <e610718166120379517e80d1a7aa12d60294209b.1387276076.git.minovotn@redhat.com>
From: Miroslav Rezanina <mrezanin@redhat.com>
Date: Tue, 19 Nov 2013 10:59:58 +0100
Subject: [PATCH 02/16] ksmtuned: Fix committed_memory when no qemu-kvm
 running

RH-Author: Miroslav Rezanina <mrezanin@redhat.com>
Message-id: <7b1d1eb263a20f4637e24fc923a3222b8a251c01.1384844315.git.mrezanin@redhat.com>
Patchwork-id: 55754
O-Subject: [RHEL7 qemu-kvm PATCH 2/3] ksmtuned: Fix committed_memory when no qemu-kvm running
Bugzilla: 1027418
RH-Acked-by: Michal Novotny <minovotn@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>

From: Miroslav Rezanina <mrezanin@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1027418

When no qemu-kvm is running, committed_memory returns empty string instead
of 0.
This is fix backported from fedora.

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 redhat/ksmtuned | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 redhat/ksmtuned | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/redhat/ksmtuned b/redhat/ksmtuned
index 91ba03b..320ce74 100644
--- a/redhat/ksmtuned
+++ b/redhat/ksmtuned
@@ -73,7 +73,7 @@ committed_memory () {
     pidlist=$(pgrep -d ' ' -- '^qemu(-kvm|:.{1,11})$')
     if [ -n "$pidlist" ]; then
         ps -p "$pidlist" -o rsz=
-    fi | awk '{ sum += $1 }; END { print sum }'
+    fi | awk '{ sum += $1 }; END { print 0+sum }'
 }
 
 free_memory () {
-- 
1.7.11.7

